LEAF_Request_Portal/js/formQuery.js

```diff
@@ -227,72 +183,57 @@ var LeafFormQuery = function () {
 
   /**
    * Add extra parameters to the end of the query API URL
-   * @param string params
+   * @param {string} params
    */
-  function setExtraParams(params) {
+  function setExtraParams(params = "") {
     extraParams = params;
   }
 
   /**
-   * @param funct - Success callback (see format for jquery ajax success)
+   * @param {function} funct - Success callback (see format for jquery ajax success)
    * @memberOf LeafFormQuery
    */
-  function onSuccess(funct) {
+  function onSuccess(funct = {}) {
     successCallback = funct;
   }
 
   /**
    * onProgress assigns a callback to be called on every getBulkData() iteration
-   * @param funct - funct(int Progress). Progress is the number of records that have been processed
+   * @param {function} funct - funct(int Progress). Progress is the number of records that have been processed
    * @memberOf LeafFormQuery
    */
-  function onProgress(funct) {
+  function onProgress(funct = {}) {
     progressCallback = funct;
   }
 
   /**
    * Execute search query in chunks
-   * @param limitOffset Used in subsequent recursive calls to track current offset
+   * @param {number} limitOffset Used in subsequent recursive calls to track current offset
    * @returns Promise resolving to query response
    * @memberOf LeafFormQuery
    */
-  let results = {};
-  let batchSize = 500;
-  function getBulkData(limitOffset) {
-    if (limitOffset == undefined) {
-      limitOffset = 0;
-    }
-    if (limitOffset == 0) {
+  function getBulkData(limitOffset = 0) {
+    if (limitOffset === 0) {
       results = {};
     }
     limitOffset = parseInt(limitOffset);
-
     query.limit = batchSize;
     query.limitOffset = limitOffset;
 
     let el = document.createElement("div");
     el.innerHTML = JSON.stringify(query);
-    let queryUrl = el.innerText;
-
-    let dataType = "json";
-    let urlParamJSONP = "";
-    if (useJSONP) {
-      dataType = "jsonp";
-      urlParamJSONP = "&format=jsonp";
-    }
 
+    const queryUrl = el.innerText;
+    const dataType = useJSONP ? "jsonp" : "json";
+    const urlParamJSONP = useJSONP ? "&format=jsonp" : "";
     return $.ajax({
```

Comment:
Ok.  Two problems here.  I know they are inherited, but we need to fix them.  One, it's returning an ajax call.  Big no-no.  It's one thing to return a promise, as hopefully what's calling it knows what to do with a promise.  Quite another to return something that can return literally anything.  I saw the below as well.  Additionally, not seeing a fail-catch.

Reply:
I've tried this in various ways and there doesn't seem to be an easy way to pull these out.  The query execute method is exposed for use in custom reports.  Depending on how the query is set up, the return value is expected to be a jq ajax call.  The other path execute could go is to a call to getBulkData, which returns recursive calls to itself (and its ajax call also behaves differently on success).